c7n/resources/sagemaker.py

```diff
@@ -397,13 +397,153 @@ def process(self, resources):
             list(w.map(self.process_instance, resources))
 
 
+@resources.register('sagemaker-model')
+class Model(QueryResourceManager):
+    class resource_type(object):
+        service = 'sagemaker'
+        enum_spec = ('list_models', 'Models', None)
+        detail_spec = (
+            'describe_model', 'ModelName',
+            'ModelName', None)
+        id = 'ModelArn'
+        name = 'ModelName'
+        date = 'CreationTime'
+        dimension = None
+        filter_name = None
+
+    filter_registry = filters
+    permissions = ('sagemaker:ListTags',)
+
+    def augment(self, resources):
+        def _augment(r):
+            client = local_session(self.session_factory).client('sagemaker')
```

Comment:
creating clients is fairly expensive its better to just create the client up one scope and then reference it inside the _augment func, client instances are thread safe.

Reply:
@kapilt need to revisit this augment... failed to realize that it prevents the details_spec from getting pulled. Need to use `super().augment()` before the _augment call... will be correcting the other resources.

Reply:
hmmm.. how it did pass tests then?